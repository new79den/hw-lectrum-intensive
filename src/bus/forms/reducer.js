import { combineForms } from 'react-redux-form'

export default combineForms({

    search: {
        text: '',
    },

}, 'forms')