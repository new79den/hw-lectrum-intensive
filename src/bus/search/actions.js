import types from "./types";

export const taskAction = Object.freeze({
    changeText: (text) => ({
        type: types.CHANGE_TEXT,
        payload: text
    })
});