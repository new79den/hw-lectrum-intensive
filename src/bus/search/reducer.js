import types from './types';

const initialState = '';

export default (state = initialState, action) => {
    switch (action.type) {
        case types.CHANGE_TEXT :
            return action.payload;
        default : return state
    }
}