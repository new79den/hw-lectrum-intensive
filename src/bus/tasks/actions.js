import types from './types';

export const taskAction = Object.freeze({
    fetchTasks: () => ({
        type: types.FETCH_TASKS,
    }),
    fetchTasksSuccess: (tasks) => ({
        type:    types.FETCH_TASKS_SUCCESS,
        payload: tasks,
    }),
    fetchTasksFail: (error) => ({
        type:    types.FETCH_TASKS_FAIL,
        payload: error,
        error:   true,
    }),

    addTask: (text) => ({
        type:    types.ADD_TASK,
        payload: text,
    }),
    addTaskSuccess: (task) => ({
        type:    types.ADD_TASK_SUCCESS,
        payload: task,
    }),
    addTaskFail: (error) => ({
        type:    types.ADD_TASK_FAIL,
        payload: error,
        error:   true,
    }),

    editTask: (task) => ({
        type:    types.EDIT_TASK,
        payload: task,
    }),
    editTaskSuccess: (task) => ({
        type:    types.EDIT_TASK_SUCCESS,
        payload: task,
    }),
    editTaskFail: (error) => ({
        type:    types.EDIT_TASK_FAIL,
        payload: error,
        error:   true,
    }),

    deleteTask: (id) => ({
        type:    types.DELETE_TASK,
        payload: id,
    }),
    deleteTaskSuccess: (id) => ({
        type:    types.DELETE_TASK_SUCCESS,
        payload: id,
    }),
    deleteTaskFail: (error) => ({
        type:    types.DELETE_TASK_FAIL,
        payload: error,
        error:   true,
    }),

    chooseAllTask: (status) => ({
        type:    types.CHOOSE_ALL_TASKS,
        payload: status,
    }),
    chooseAllTasksSuccess: (tasks) => ({
        type:    types.CHOOSE_ALL_TASKS_SUCCESS,
        payload: tasks,
    }),
    chooseAllTaskFail: (error) => ({
        type:    types.CHOOSE_ALL_TASKS_FAIL,
        payload: error,
        error:   true,
    }),
});
