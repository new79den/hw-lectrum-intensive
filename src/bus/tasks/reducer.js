//Core
import {List, fromJS} from "immutable";
// Instruments
import types from './types';

const initialState = List([]);

export default (state = initialState, action) => {
    switch (action.type) {

        case types.FETCH_TASKS_SUCCESS:
            return fromJS(action.payload);

        case types.ADD_TASK_SUCCESS:
            return state.unshift(fromJS(action.payload));

        case types.DELETE_TASK_SUCCESS:
            return state.filter(el => el.get('id') !== action.payload);

        case types.EDIT_TASK_SUCCESS:
            const indexOfListToUpdate = state.findIndex(el => el.get('id') === action.payload.id);
            return state.update(indexOfListToUpdate, el => {
                return el.set("message", action.payload.message)
                    .set('completed', action.payload.completed)
                    .set('favorite', action.payload.favorite)
                    .set('modified', action.payload.modified);
            });

        case types.CHOOSE_ALL_TASKS_SUCCESS:
            return state.merge(fromJS(action.payload));

        default:
            return state
    }
}