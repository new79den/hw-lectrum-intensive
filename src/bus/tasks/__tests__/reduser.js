// Core
import { taskAction } from '../actions';
import  reducer  from '../reducer';
import { fromJS } from 'immutable';


describe('tasks reducer:', () => {
    test('should return initial state', () => {
        expect(reducer(void 0, {})).toMatchSnapshot();
    });
    test('should handle fetchTasksSuccess action', () => {
        expect(reducer(void 0, taskAction.fetchTasksSuccess(__.tasks))).toMatchSnapshot();
    });
    test('should handle addTaskSuccess action', () => {
        expect(reducer(fromJS(__.tasks), taskAction.addTaskSuccess(__.task))).toMatchSnapshot();
    });
    test('should handle deleteTaskSuccess action', () => {
        expect(reducer(fromJS(__.tasks), taskAction.deleteTaskSuccess(__.tasks[0].id))).toMatchSnapshot();
    });
    test('should handle editTaskSuccess action', () => {
        expect(reducer(fromJS(__.tasks), taskAction.editTaskSuccess(__.updateTask))).toMatchSnapshot();
    });
    test('should handle chooseAllTasksSuccess action', () => {
        expect(reducer(fromJS(__.tasks), taskAction.chooseAllTasksSuccess(__.chooseAllTasks))).toMatchSnapshot();
    });
});