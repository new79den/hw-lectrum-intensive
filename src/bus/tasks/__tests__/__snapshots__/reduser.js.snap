// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tasks reducer: should handle addTaskSuccess action 1`] = `
Immutable.List [
  Immutable.Map {
    "message": "the request has succeeded",
    "data": Immutable.Map {
      "id": "5a7f136331a5d90001271638",
      "message": "Hello",
      "completed": false,
      "favorite": false,
      "created": "2018-02-10T15:44:35.284Z",
    },
  },
  Immutable.Map {
    "id": "5a7f136231a5d90001271637",
    "message": "Hello!",
    "completed": true,
    "favorite": false,
    "created": "2018-02-10T15:44:34.624Z",
    "modified": "2018-02-10T16:01:12.406Z",
  },
  Immutable.Map {
    "id": "5a7f136131a5d90001271636",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:33.675Z",
  },
  Immutable.Map {
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:32.959Z",
  },
]
`;

exports[`tasks reducer: should handle chooseAllTasksSuccess action 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "5a7f136231a5d90001271637",
    "message": "Hello!",
    "completed": true,
    "favorite": false,
    "created": "2018-03-10T15:44:34.624Z",
    "modified": "2018-04-10T16:01:12.406Z",
  },
  Immutable.Map {
    "id": "5a7f136131a5d90001271636",
    "message": "Hello",
    "completed": true,
    "favorite": false,
    "created": "2018-02-10T15:44:33.675Z",
    "modified": "2018-04-10T16:01:12.406Z",
  },
  Immutable.Map {
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
    "completed": true,
    "favorite": false,
    "created": "2018-02-10T15:44:32.959Z",
    "modified": "2018-04-10T16:01:12.406Z",
  },
]
`;

exports[`tasks reducer: should handle deleteTaskSuccess action 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "5a7f136131a5d90001271636",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:33.675Z",
  },
  Immutable.Map {
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:32.959Z",
  },
]
`;

exports[`tasks reducer: should handle editTaskSuccess action 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "5a7f136231a5d90001271637",
    "message": "Hello!",
    "completed": true,
    "favorite": false,
    "created": "2018-02-10T15:44:34.624Z",
    "modified": "2018-02-10T16:01:12.406Z",
  },
  Immutable.Map {
    "id": "5a7f136131a5d90001271636",
    "message": "test update",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:33.675Z",
    "modified": "2018-02-10T16:01:12.406Z",
  },
  Immutable.Map {
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:32.959Z",
  },
]
`;

exports[`tasks reducer: should handle fetchTasksSuccess action 1`] = `
Immutable.List [
  Immutable.Map {
    "id": "5a7f136231a5d90001271637",
    "message": "Hello!",
    "completed": true,
    "favorite": false,
    "created": "2018-02-10T15:44:34.624Z",
    "modified": "2018-02-10T16:01:12.406Z",
  },
  Immutable.Map {
    "id": "5a7f136131a5d90001271636",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:33.675Z",
  },
  Immutable.Map {
    "id": "5a7f136031a5d90001271635",
    "message": "Hello",
    "completed": false,
    "favorite": false,
    "created": "2018-02-10T15:44:32.959Z",
  },
]
`;

exports[`tasks reducer: should return initial state 1`] = `Immutable.List []`;
